### PR TITLE
Enable engineering mode

### PR DESCRIPTION
**Pull request by triplem**

### CODE REVIEW

**Comment by triplem** [comment]
One note, to configure the directory where the corresponding file should be put, is configured in the log-configuration. the log-directory is therefor the same as the directory this file needs to be put in.

File: app/src/main/java/info/nightscout/androidaps/MainApp.java [hunk 1]
```diff
@@ -195,6 +202,16 @@ public void run() {
             }
         }).start();
 
+        String extFilesDir = this.getLogDirectory();
+        File engineeringModeSemaphore = new File(extFilesDir,"engineering_mode");
+
+        engineeringMode = engineeringModeSemaphore.exists() && engineeringModeSemaphore.isFile();
+        devBranch = BuildConfig.VERSION.contains("dev");
+
+        if (!isDevModeOrRelease()) {
```
Comment (jotomo) [inline app/src/main/java/info/nightscout/androidaps/MainApp.java]: The name makes it easy to confuse dev-branch and engineering-mode. Should be `isEngineeringModeOrRelease`.

File: app/src/main/java/info/nightscout/androidaps/MainActivity.java [hunk 2]
```diff
@@ -400,6 +400,8 @@ public void onClick(DialogInterface dialog, int which) {
                                     builder.setIcon(R.mipmap.blueowl);
                                 String message = "Build: " + BuildConfig.BUILDVERSION + "\n";
                                 message += MainApp.sResources.getString(R.string.configbuilder_nightscoutversion_label) + " " + ConfigBuilderPlugin.nightscoutVersionName;
+                                if (MainApp.engineeringMode)
+                                    message += MainApp.gs(R.string.engineering_mode_enabled);
```
Comment (jotomo) [inline app/src/main/java/info/nightscout/androidaps/MainActivity.java]: I forgot to prepend a newline here.
image

**Comment by jotomo** [comment]
Tested release version, dev version, engineering mode enabled. Works.

**Comment by jotomo** [comment]
:+1:

**Comment by AdrianLxM** [comment]
@MilosKozak Anyone of us hit merge?

**Pull request merged**